scenarios/aws/microVMs/microvms/run.go

```diff
@@ -73,7 +73,7 @@ func newEC2Instance(awsEnv aws.Environment, name, ami, arch, instanceType, keyPa
 
 	instance, err := awsEc2.NewInstance(awsEnv.Ctx, awsEnv.Namer.ResourceName(name), &awsEc2.InstanceArgs{
 		Ami:                 pulumi.StringPtr(ami),
-		SubnetId:            pulumi.StringPtr(awsEnv.DefaultSubnets()[0]),
+		SubnetId:            awsEnv.RandomSubnets().Index(pulumi.Int(0)),
```

Comment:
@usamasaqib As you're tweaking the subnets in your test, does that work for you?

Reply:
We are tweaking the subnets for two reason.
1. The default subnet in datadog-agent-qa had capacity issues for the metal instance we were spawning. Therefore, we passed different subnets to target.
2. If the stack creation fails due to capacity shortage, we retry the stack with a different subnet.

We haven't tried addressing the IP problem by changing subnets.